### PR TITLE
Fix almost all remaining warnings from the toolchain update.

### PR DESCRIPTION
**Pull request by cbiffle**
This fixes everything but Hiffy's use of unsafe (or anything anybody has added since I started on this branch).

This change turns Clippy back on to prevent regressions.

### CODE REVIEW

File: task/monorail-server/src/server.rs [hunk 1]
```diff
@@ -365,9 +366,7 @@ impl<'a, R: Vsc7448Rw> idl::InOrderMonorailImpl for ServerImpl<'a, R> {
                 if let Some(Err(e)) = self.bsp.phy_fn(port, |phy| {
                     phy.read(phy::STANDARD::INTERRUPT_STATUS())
                 }) {
-                    return Err(e)
-                        .map_err(MonorailError::from)
-                        .map_err(RequestError::from);
+                    return Err(RequestError::from(MonorailError::from(e)));
```
Comment (mkeeter) [inline task/monorail-server/src/server.rs:369]: how dare you, clippy

**Review by mkeeter — APPROVED** [review]

File: task/monorail-server/src/server.rs (hunk 1 above)
Comment (cbiffle) [reply task/monorail-server/src/server.rs:369]: Clippy says to meet it behind the school at 3pm and, quote, "let's finish this"

**Review by hawkw — APPROVED** [review]

File: drv/sidecar-seq-server/src/main.rs [hunk 2]
```diff
@@ -834,7 +834,7 @@ fn main() -> ! {
                 .load_bitstream(AUXFLASH.get_task_id())
             {
                 Err(e) => {
-                    let code = u32::try_from(e).unwrap_lite();
```
Comment (labbott) [inline drv/sidecar-seq-server/src/main.rs:837]: Everytime an `unwrap` gets removed a microcontroller gets its wings

File: task/control-plane-agent/src/mgs_psc.rs [hunk 3]
```diff
@@ -46,8 +46,6 @@ static UPDATE_MEMORY: UpdateBuffer = UpdateBuffer::new();
 
 pub(crate) struct MgsHandler {
     common: MgsCommon,
-    sp_update: SpUpdate,
-    rot_update: RotUpdate,
```
Comment (labbott) [inline task/control-plane-agent/src/mgs_psc.rs:50]: Did those get folded into some default somewhere?

Comment (mkeeter) [reply task/control-plane-agent/src/mgs_psc.rs:50]: Yes, #1731

**Review by labbott — APPROVED** [review]
Long live clippy

**Pull request merged**